### PR TITLE
Remove no longer needed rest client modules from registry and refactor rest client to use new repo

### PR DESCRIPTION
**Pull request by carlesarnal**
This PR refactors our rest clients to use the new repo with common rest clients features.

- The tenant manager has been adapted to use the new auth capabilities and features available in the new repo.
- The registry client has been fully adapted to use all the new capabilities and features available in the new repo.

Note that this PR will fail until the new [repo](https://github.com/Apicurio/apicurio-common-rest-client) artifacts are public.

### CODE REVIEW

**Review by famartinrh — APPROVED** [review]
LGTM

**Comment by carlesarnal** [comment]
The new repo is public so this should work now. This pr also removes any reference to keycloak and the apache HTTP client so any issues with native compilation caused by Keycloak should be fixed now.

**Comment by famartinrh** [comment]
@EricWittmann @carlesarnal do we merge this?

**Comment by carlesarnal** [comment]
I'm happy to merge this. I was just waiting for @EricWittmann to take a look at it.

**Comment by EricWittmann** [comment]
@carlesarnal Sorry this wasn't done earlier - now there are conflicts. :( :(  Can you fix the conflicts?

**Comment by carlesarnal** [comment]
> @carlesarnal Sorry this wasn't done earlier - now there are conflicts. :( :( Can you fix the conflicts?

No problem, actually I've released a new version of the library and I need to update this before merging.

File: app/src/main/resources/application.properties [hunk 1]
```diff
@@ -99,8 +99,10 @@ registry.multitenancy.base.path=t
 registry.organization-id.claim-name=${ORGANIZATION_ID_CLAIM:rh-org-id}
 registry.tenant.manager.url=${TENANT_MANAGER_URL:http://localhost:8585}
 
-registry.tenant.manager.auth.url=${TENANT_MANAGER_AUTH_URL:http://localhost:8090/auth}
 registry.tenant.manager.auth.realm=${TENANT_MANAGER_REALM:registry}
+registry.tenant.manager.auth.url=${TENANT_MANAGER_AUTH_URL:http://localhost:8090/auth}
+registry.tenant.manager.auth.url.configured=${TENANT_MANAGER_TOKEN_ENDPOINT:${registry.tenant.manager.auth.url}/realms/${registry.tenant.manager.auth.realm}/protocol/openid-connect/token}
```
Comment (famartinrh) [inline app/src/main/resources/application.properties:104]: great job!

File: multitenancy/tenant-manager-client/src/main/java/io/apicurio/multitenant/client/TenantManagerClientImpl.java [hunk 2]
```diff
@@ -15,70 +15,71 @@
...
-        this.auth = auth;
+        this.client = new JdkHttpClient(endpoint, configs, auth, new TenantManagerClientErrorHandler());
+    }
+
+    public TenantManagerClientImpl(Vertx vertx, String baseUrl, Map<String, Object> configs, Auth auth) {
+        if (!baseUrl.endsWith("/")) {
+            baseUrl += "/";
+        }
+        final String endpoint = baseUrl;
+        this.mapper = new ObjectMapper();
```
Comment (famartinrh) [inline multitenancy/tenant-manager-client/src/main/java/io/apicurio/multitenant/client/TenantManagerClientImpl.java:81]: this should be `this.mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);`

**Review by famartinrh — APPROVED** [review]
LGTM, just a minor change that we can do later

great job!

File: multitenancy/tenant-manager-client/src/main/java/io/apicurio/multitenant/client/TenantManagerClientImpl.java (hunk 2 above)
Comment (carlesarnal) [reply multitenancy/tenant-manager-client/src/main/java/io/apicurio/multitenant/client/TenantManagerClientImpl.java:81]: Oh yes, sorry, this is probably a rebase issue.

**Comment by carlesarnal** [comment]
Last comment addressed. I think his must be deployed along with [this](https://github.com/bf2fc6cc711aee1a0c2a/srs-fleet-manager/pull/120).

**Comment by famartinrh** [comment]
is there any hard requirement to deploy those two things together?

**Comment by carlesarnal** [comment]
This pr changes a lot of dependencies in the tenant manager client. I don't think it's a hard requirement to deploy them together but something desirable. That said, while this can be deployed along that will not work the other way around, the other pr cannot be merged without this one. The other pr is just the equivalent of this one but for the fleet manager.

**Comment by famartinrh** [comment]
ok, let's try to deploy this PR first, and then (after fixing tests in the other PR) deploy that one

**Pull request merged**